### ISSUE TITLE
Game crashing randomly after installing RU

### ISSUE THREAD

**Issue by P26601** [UHQ-GAMES-MODS/REGIONS_UNEXPLORED_FORGE#131]
**Describe the bug**
Game seems to crash randomly when exploring. Save can be re-loaded but game crashes again after a few minutes

**Crash Logs**
https://pastebin.com/FL2M2eFM

**Version**
Minecraft Version: 1.20.4

Modloader Version: Fabric Loader 0.15.7

Mod Version: 0.5.4

**Comment by KirboSoftware** [comment]
I've experienced the same crash as well at some point on my own server with the latest version, for now I've just installed [neruina](https://www.curseforge.com/minecraft/mc-mods/neruina) and seemed to fix my issues. However, now that someone else also experienced it. I will do a deep dive! Thanks for reporting and I will do my best to fix it

**Comment by P26601** [comment]
Thanks! Unfortunately my game keeps crashing even with neruina. I've installed mixintrace for a more detailed crash report and there seems to be a conflict with yung's better desert temples ([new crash report](https://pastebin.com/kqAP9aXz), lines 118-133), but I really don't know anything about this stuff so I'm not sure lmao

**Comment by KirboSoftware** [comment]
Hmmm, I will do more tests if it's a compat issue with YUNG's api. Thanks for letting me know and I will run more tests with this!

**Comment by purejosh** [comment]
I can recreate this without any YUNG's mods. Exactly this error, on 1.20.1 Fabric.

**Comment by UHQ-GAMES** [comment]
> I can recreate this without any YUNG's mods. Exactly this error, on 1.20.1 Fabric.

can you also send your log? i know you said its the same error but if you could send it anyways anything can help with a weird issue like this

**Comment by KirboSoftware** [comment]
UHQ found the source and will be included in the next patch 👍

**Comment by purejosh** [comment]
> > I can recreate this without any YUNG's mods. Exactly this error, on 1.20.1 Fabric.
> 
> can you also send your log? i know you said its the same error but if you could send it anyways anything can help with a weird issue like this

https://gist.github.com/purejosh/7f62bd4d1f6f8e7ee34a8cf56cac6d5c

**Comment by purejosh** [comment]
> UHQ found the source and will be included in the next patch 👍

Oh, good! I'm glad to hear it. There's something really weird though.... it crashes my client when I run the world from it, but if I join my Fabric server with my client, the server doesn't crash at all. Why would this crash clients, but not servers? 

But also... it crashed a client of mine's server, too? Did I just get really lucky?

**Comment by purejosh** [comment]
Could this be related, too? 
```
[14:58:36] [ForkJoinPool.commonPool-worker-16/WARN]: [RER] Failed to parse loot table 'regions_unexplored:blocks/windswept_grass': 
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[?:?]
	at java.util.Objects.checkIndex(Objects.java:361) ~[?:?]
	at java.util.ArrayList.get(ArrayList.java:427) ~[?:?]
	at com.google.gson.JsonArray.get(JsonArray.java:228) ~[gson-2.10.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootCondition(LootDisplay.java:200) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootEntryJson(LootDisplay.java:248) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootEntryAlternativesJson(LootDisplay.java:102) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootEntryJson(LootDisplay.java:220) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootPoolJson(LootDisplay.java:445) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.munchLootSupplierJson(LootDisplay.java:478) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.getOutputs(LootDisplay.java:490) ~[rer-2.9.0.jar:?]
	at uk.me.desert_island.rer.rei_stuff.LootDisplay.getOutputEntries(LootDisplay.java:66) ~[rer-2.9.0.jar:?]
	at me.shedaniel.rei.plugin.client.runtime.FilteredStacksVisibilityHandler.checkHiddenStacks(FilteredStacksVisibilityHandler.java:105) ~[RoughlyEnoughItems-12.0.684-fabric.jar:?]
	at me.shedaniel.rei.plugin.client.runtime.FilteredStacksVisibilityHandler.lambda$cacheExisting$0(FilteredStacksVisibilityHandler.java:90) ~[RoughlyEnoughItems-12.0.684-fabric.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290) ~[?:?]
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
```

**Comment by KirboSoftware** [comment]
We already identified the issue and will be fixed soon:tm: when we have time to patch all versions

**Comment by UHQ-GAMES** [comment]
fixed in 0.5.5 13edeeab0061531f96e27e525b880df1e759f3b8
